### ISSUE TITLE
Threshold test fails on 32-bit

### ISSUE THREAD

**Issue by stefanv** [scikit-image/scikit-image#2527]
```
======================================================================
FAIL: skimage.filters.tests.test_thresholding.test_threshold_minimum
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/venv/local/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/venv/local/lib/python2.7/site-packages/skimage/filters/tests/test_thresholding.py", line 286, in test_threshold_minimum
    assert threshold == 76
AssertionError: 
-------------------- >> begin captured logging << --------------------
PIL.PngImagePlugin: DEBUG: STREAM 'IHDR' 16 13
PIL.PngImagePlugin: DEBUG: STREAM 'pHYs' 41 9
PIL.PngImagePlugin: DEBUG: STREAM 'cHRM' 62 32
PIL.PngImagePlugin: DEBUG: 'cHRM' 62 32 (unknown)
PIL.PngImagePlugin: DEBUG: STREAM 'IDAT' 106 114106
--------------------- >> end captured logging << ---------------------
```

**Comment by cgohlke** [comment]
FWIW, I can not reproduce this issue, https://github.com/scikit-image/scikit-image/issues/2529, or https://github.com/scikit-image/scikit-image/issues/2528 on Windows with scikit_image-0.13.dev0 (current git master), numpy-1.11.3+mkl, scipy-0.18.1, matplotlib-2.0.0, Pillow-4.0.0, pyamg-3.2.1, PyWavelets-0.5.1, and dask-0.14.0. All tests pass on all supported Python platforms.

**Comment by soupault** [comment]
What are our options for testing and debugging in 32-bit environments? To the best of my knowledge, none of AWS, Travis-CI, or Appveyor support 32-bit instances.

**Comment by stefanv** [comment]
If you run a Debian container, you can set up a 32-bit chroot:

https://debian-administration.org/article/356/Installing_apps_in_a_32-bit_chroot_in_AMD64_Debian_system

If you build a Docker image with those steps done, please share!

**Comment by stefanv** [comment]
@cgohlke Thanks for testing; that's encouraging.

**Comment by matthew-brett** [comment]
I found this error running the tests on the standard travis-ci machines, with build and tests inside a 32-bit container.  On your local Linux machine, try something like:

```
cd scikit-image
git checkout master
git pull
docker pull matthewbrett/trusty:32
docker run --rm -ti -e PYTHON_VERSION=2.7 -v $PWD:/io matthewbrett/trusty:32 /bin/bash
```

Inside the container that starts up:

```
pip install Cython numpy nose
# Matplotlib i686 wheels are not on pypi
pip install -f https://5cf40426d9f06eb7461d-6fe47d9331aba7cd62fc36c7196769e4.ssl.cf2.rackcdn.com matplotlib
cd /io  # now in scikit-image directory from host
pip install -e .
nosetests skimage/filters/tests/test_thresholding.py:test_threshold_minimum
```

The scikit-image build takes a while, but other steps should be reasonably quick, with a decent internet connection.

**Comment by stefanv** [comment]
The bug is due to a difference in behavior in `scipy.ndimage.filters.uniform_filter1d` on i686 vs x86_64.

**Comment by stefanv** [comment]
This issue may be related: https://github.com/scipy/scipy/issues/6930

**Comment by stefanv** [comment]
A first attempt at a fix: https://github.com/scikit-image/scikit-image/pull/2549

**Comment by soupault** [comment]
@matthew-brett Thanks a ton, Matthew! Works like a charm!

**Comment by soupault** [comment]
I do not see this error anymore. @jni , is fixed via #2549, I assume?

**Comment by jni** [comment]
@soupault I believe so!
